src/java.base/share/classes/sun/net/www/protocol/file/FileURLConnection.java

```diff
@@ -210,6 +210,8 @@ public synchronized InputStream getInputStream()
     /* since getOutputStream isn't supported, only read permission is
      * relevant
      */
+    @Override
+    @Deprecated(since = "25", forRemoval = true)
```

Comment:
Is this annotation required or more as a reminder that the superclass method is deprecated?

Reply:
I think it's needed because the superclass method is deprecated for removal. Will double check.

Reply:
Yes - it is needed. If you override a deprecated method you need to carry the deprecation annotation over. Here is what I got when I tried to remove the annotation on `HttpURLConnection::getPermission()` (the same error occurred on any subclass where I attempted to remove the annotation):
```
java/net/HttpURLConnection.java:615: warning: [dep-ann] deprecated item is not annotated with @Deprecated
    public Permission getPermission() throws IOException {
                      ^
error: warnings found and -Werror specified
```